doc/guides/west/build-flash-debug.rst

```diff
@@ -86,6 +86,18 @@ To use another build directory, use ``--build-dir`` (or ``-d``)::
 
   west build -b <BOARD> --build-dir path/to/build/directory
 
+To change the default build directory from ``build`` to a user-defined format,
+use the ``dir-fmt`` configuration option::
+
+  west config build.dir-fmt "build/{board}/{source_dir}"
```

Comment:
Is it just me, or would changing this to `{app}` and making the result be the basename of the source directory be easier for most people to use?

```
$ cd ~/zephyrproject
$ west config build.dir-fmt build/{board}/{app}
$ west build -b reel_board zephyr/samples/hello_world  # makes build/hello_world/reel_board 
$ west build -b qemu_x86 zephyr/samples/hello_world    # makes build/hello_world/qemu_x86
```

Especially if I want faster tab completion on `west flash -d`.

Reply:
Sure, no objections. `source_dir` is consistent with other places, but if you think `{app}` is clearer I have no issues changing it

Reply:
Oh sorry, I now understand. See my comment here: https://github.com/zephyrproject-rtos/zephyr/pull/16636#issuecomment-499104299

Reply:
I'm OK with both and would rather have a separate {app} than {source_dir[:1]}.

Note that at least for upstream samples, app names tend to be unique. These are the only dupes I could find:

```
      2 ap
      2 echo
      3 philosophers
      2 remote
      2 sensor
      2 timer_synchronization
```

So unless you're literally building all the samples, {app} is not going to collide.